crypto/evp_extra/p_kem.c

```diff
@@ -55,6 +97,78 @@ static int pkey_kem_keygen(EVP_PKEY_CTX *ctx, EVP_PKEY *pkey) {
   return 1;
 }
 
+static int pkey_kem_encapsulate_deterministic(EVP_PKEY_CTX *ctx,
+                                              uint8_t *ciphertext,
+                                              size_t  *ciphertext_len,
+                                              uint8_t *shared_secret,
+                                              size_t  *shared_secret_len,
+                                              const uint8_t *seed,
+                                              size_t *seed_len) {
+  KEM_PKEY_CTX *dctx = ctx->data;
+  const KEM *kem = dctx->kem;
+  if (kem == NULL) {
+    if (ctx->pkey == NULL) {
+      OPENSSL_PUT_ERROR(EVP, EVP_R_NO_PARAMETERS_SET);
+      return 0;
+    }
+    kem = KEM_KEY_get0_kem(ctx->pkey->pkey.kem_key);
+  }
+
+  // Caller is getting parameter values.
+  if (ciphertext == NULL && shared_secret == NULL && seed == NULL) {
+    *ciphertext_len = kem->ciphertext_len;
+    *shared_secret_len = kem->shared_secret_len;
+    *seed_len = kem->encaps_seed_len;
```

Comment:
Thinking about this more, all of these in/out sizes also need null checks before writing to them. Same for keygen.

Reply:
Great call out, as the caller will use seed_len as a length check, better make it writable. Added test coverage also for these checks in [fa6e270](https://github.com/aws/aws-lc/pull/1578/commits/fa6e270f9b487fb1ffc82716a4c88720ca1bbf86)